astor/code_gen.py

```diff
@@ -81,34 +112,108 @@ def set_precedence(value, *nodes):
 set_precedence = precedence_setter()
 
 
+def conditional(*params):
+    """ If the last param evaluates to None, just return
+        an empty tuple
+    """
+    return params if params[-1] is not None else ()
+
+
+def comma_group(items, trailing=False,
+                # Constants
+                set_precedence=set_precedence,
+                Comma=Precedence.Comma, enumerate=enumerate):
+    """ Print a group of 0 or more items, with commas between,
+        and an optional trailing comma.
+    """
+    set_precedence(Comma, *items)
+    yield (((', ', item) if idx else item)
+                for idx, item in enumerate(items))
+    if trailing:
+        yield ','
+
+
+def else_body(node,
+              # Constants
+              If=ast.If, set_precedence=set_precedence,
```

Comment:
Out of curiosity, are these optimizations still produce big speedups in a modern Python? We don't use this technique in the Python stdlib anymore because some core developers said that the performance gain is no longer huge, but I don't remember seeing any benchmark.